### PR TITLE
Epic 7 - Anastasiia Borshch

### PR DESCRIPTION
**Pull request by nastiaborsch**

### CODE REVIEW

**Comment by okohuch** [comment]
стоп

**Comment by izagor** [comment]
+ 12-19-2023

File: ai_12/anastasiia_borshch/Epic 7/control_work_task1_anastasiia_borsch.cpp [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+const int MY_CONSTANT = 959;
+
+// Математична функція для заповнення матриці
+int mathematicalFunction(int x, int y) {
+    // Умова завдання 2: додавання константи до координат елемента
+    return x + y + MY_CONSTANT;
+}
+
+// Функція для створення та повернення матриці 10x10
+int** createMatrix() {
```
Comment (fm2lpnu) [inline ai_12/anastasiia_borshch/Epic 7/control_work_task1_anastasiia_borsch.cpp:16]: Task 1 (a,b) is not implemented according to requirements.
One of two options (a,b) is covered.

File: ai_12/anastasiia_borshch/Epic 7/control_work_task1_anastasiia_borsch.cpp [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    std::cout << "Original Matrix:" << std::endl;
+    printMatrix(resultMatrix, 10, 10);
+
+    // Значення для другого параметра (можна ввести з консолі)
+    float secondParameter;
+    std::cout << "Enter a floating-point value for the second parameter: ";
+    std::cin >> secondParameter;
+
+    // Умова завдання 2: виклик функції для сортування рядків матриці
+    bubbleSortRows(resultMatrix, 10, 10);
```
Comment (fm2lpnu) [inline ai_12/anastasiia_borshch/Epic 7/control_work_task1_anastasiia_borsch.cpp:197]: Task 2 mismatch with requirements:
Sort should be done inside overloaded function

File: ai_12/anastasiia_borshch/Epic 7/control_work_task1_anastasiia_borsch.cpp [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+        }
+    }
+}
+
+// Функція для отримання суми 3-4 рандомних елементів у стовпці
+int getColumnSum(int** matrix, int rows, int col) {
+    int sum = 0;
+
+    // Отримання 3-4 рандомних елементів у стовпці
+    for (int i = 0; i < std::min(4, rows); ++i) {
```
Comment (fm2lpnu) [inline ai_12/anastasiia_borshch/Epic 7/control_work_task1_anastasiia_borsch.cpp:58]: Task 3 Mismatch with requirements.
Random elements index is achieved using `rand()`

**Review by fm2lpnu — APPROVED** [review]

**Pull request merged**